src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -149,6 +155,18 @@ public async Task<bool> ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem up
         }
         else
         {
+            if (checkNextCommitToProcess)
+            {
+                if (!string.IsNullOrEmpty(pr.NextCommitToProcess) && pr.NextCommitToProcess != update.SourceSha)
+                {
+                    _logger.LogInformation("Skipping update for subscription {subscriptionId} commit {updateCommit} since there's a newer commit {nextCommit}",
```

Comment:
Please improve this log statement - I believe we should at least include build IDs because that's what we work with mostly. Commits are hard to map. Also I think "since there's a newer commit" can be confusing if you find it in the logs. I would properly explain that an update with a newer build is scheduled already.

Actually in general, I think we should be storing build IDs instead of commits. There could be multiple builds from a same commit and we're after the latest applied build, not commit.